tsconfig.json

```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "charset": "utf-8",
+    "declaration": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "lib": ["ES2019", "es2020.promise"],
```

Comment:
we probably need a less opinionated tsconfig and also target the lowest common denominator (nodejs 10) I think this means ES2017

Reply:
Updated to es2017